### PR TITLE
X2000 Update

### PR DESCRIPTION
**Pull request by jfkuayue**
X2000 Update
Variable lengths for SJ X2: a loco and 5-7 cars

### CODE REVIEW

File: make.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 copy_bool = True
 
 # What is the path of your OpenTTD folder?
-openttd_path = "C:/Users/tyler/Documents/OpenTTD/newgrf"
```
Comment (2TallTyler) [inline make.py]: Changes to the makefile shouldn't be included in this PR. Ideally this would be system-agnostic, but I'm not sure quite how to do that... So just leave this change on your local branch. 🙂

File: src/x2000.nml [hunk 2]
```diff
@@ -34,54 +34,54 @@ spriteset(spriteset_x2000_2_cab_reversed, "src/gfx/x2000_2_cab_8bpp.png") {templ
...
 /* Select livery type */
 
-switch (FEAT_TRAINS, SELF, sw_x2000_livery_date, date_of_last_service) {
-	0..date(2005, 1, 1): return 0;
-	return 1;
-}
+//switch (FEAT_TRAINS, SELF, sw_x2000_livery_date, date_of_last_service) {
+//	0..date(2005, 1, 1): return 0;
+//	return 1;
+//}
```
Comment (2TallTyler) [inline src/x2000.nml]: This eliminates the automatic livery selection feature.

**Review by 2TallTyler — COMMENTED** [review]
Thanks for this.

I'm not willing to lose the automatic livery feature for this train, and have never been a fan of cargo refits for length changes (or liveries, for that matter, but variants didn't exist when I created this).

If you're up for it, you could add the different length versions as variants.

In a perfect world, I'd ditch cargo refits altogether for everything in this set and use only variants for livery changes. But I don't have the time/interest in re-coding this trainset in such a way.

File: src/x2000.nml (hunk 2 above)
Comment (jfkuayue) [reply src/x2000.nml]: Hello. Adding variants would omit those who use 12.2 and before. Are these versions really outdated? I ask for this considering those mobile players.

Comment (2TallTyler) [reply src/x2000.nml]: Updating OpenTTD is easy and free. I'm not worried about supporting versions that are well over a year old.

Comment (jfkuayue) [reply src/x2000.nml]: On iOS system, the newest version is pseudo-12.1. And it seems the variants do not work properly before 13.4.

Comment (2TallTyler) [reply src/x2000.nml]: The iOS version is unofficial and unsupported. Unless I'm missing something in the changelog, the last change to variants was in 13.1 (the engine name callback), which was released 10 April 2023. Variants were added in 13.0 so they fail to load (fatal NewGRF error) in older versions of the game. 

I am really not worried about supporting outdated versions of OpenTTD, but if you're playing on such a version, of course I understand if you don't want to create something you can't use. 😃

Comment (jfkuayue) [reply src/x2000.nml]: Hello. I just implemented variants for X2000. And it works properly now.

**Review by 2TallTyler — APPROVED** [review]
Thank you!

**Comment by jfkuayue** [comment]
Oh, and since variants instead of cargo subtypes are used, "gfx/empty.png" can be deleted now.

**Comment by 2TallTyler** [comment]
Thanks, fixed in https://github.com/2TallTyler/danish_trains/commit/26394d317fe51c8f7a94efb963f2919656794149.
